pandas/core/reshape/concat.py

```diff
@@ -250,7 +250,10 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
 
         if isinstance(objs, dict):
             if keys is None:
-                keys = sorted(objs)
+                if not isinstance(objs, OrderedDict):
+                    keys = sorted(objs)
```

Comment:
Normal dicts are ordered too in python3.6+, that must be checked too. There is a ``PY36`` constant somewhere, that you can use for this check.

Reply:
Not sure if we already took a stance on this as a project, but the orderedness of dicts in 3.6 is considered an [implementation detail](https://stackoverflow.com/a/39980744/2858145) of cpython, while it will be a language feature starting from 3.7 . So in principle in 3.6 users should not rely on it, and we should not assume they do.

@topper-123 's comment still holds I think, but for 3.7

Reply:
Well we've already relied on that implementation detail in other changes (see #19830, #19859, #19884) so I'd be +1 on relying on that here as well

Reply:
Yeah, given that Python 3.6 is at 3.6.5 now, and its's being formalised in 3.7, I think this is quite safe. They're not going to change implementation now...

Reply:
This is very simliar to in structure to ``pandas.core.common._dict_keys_to_ordered_list``. structure like this. see if we can consolidate all of these try/sorting routings.

Reply:
yes that's the point i am raising, I want a refactor here.

Reply:
So, from ```_dict_keys_to_ordered_list``` method,  i suggest change the condition```if PY36 or isinstance(mapping, OrderedDict):``` to ```if PY37 or isinstance(mapping, OrderedDict):``` then, remove ```test_constructor_dict_order_insertion```  and ```test_constructor_dict_order```. What do you think about it?